code/modules/mob/living/carbon/human/ai/brain/ai_brain_items.dm

```diff
@@ -100,20 +112,20 @@
 
 	if((slot in important_storage_slots) && istype(equipment, /obj/item/storage))
 		recalculate_containers()
-		appraise_inventory(slot == WEAR_WAIST, slot == WEAR_BACK, slot == WEAR_L_STORE, slot == WEAR_R_STORE)
+		appraise_inventory(slot == WEAR_WAIST, slot == WEAR_BACK, slot == WEAR_L_STORE, slot == WEAR_R_STORE, slot == WEAR_JACKET, slot == WEAR_BODY)
 
-	if((!primary_weapon || (primary_weapon?.w_class < equipment.w_class)) && isgun(equipment))
+	if(!primary_weapon && isgun(equipment))
```

Comment:
When we pick up our gun for the first time in life, in most cases it's a pistol spawned inside belt by equipment_preset.
Problem being - AI will select it as our primary weapon before everything else, until we go and change order in which weapons are given by certain presets
So for basic compitability with non-ai equipments i'd recommend either adding a suggested check, or compare appraisals here for both new and old weapon
```suggestion
	if(!primary_weapon && isgun(equipment) && (slot == WEAR_J_STORE))
```